charon/src/bin/charon-driver/translate/translate_bodies.rs

```diff
@@ -926,38 +926,59 @@ impl BodyTransCtx<'_, '_, '_> {
                 let then_block = self.translate_basic_block_id(*target);
                 Ok(SwitchTargets::If(if_block, then_block))
             }
-            LiteralTy::Int(int_ty) => {
-                let targets: Vec<(ScalarValue, BlockId)> = targets
+            LiteralTy::Char => {
+                let targets: Vec<(Literal, BlockId)> = targets
                     .iter()
                     .map(|(v, tgt)| {
-                        let v =
-                            ScalarValue::from_le_bytes(IntegerTy::Signed(int_ty), v.data_le_bytes);
+                        let b: u128 = u128::from_le_bytes(v.data_le_bytes);
+                        let v = Literal::char_from_le_bytes(b);
                         let tgt = self.translate_basic_block_id(*tgt);
                         (v, tgt)
                     })
                     .collect();
                 let otherwise = self.translate_basic_block_id(*otherwise);
                 Ok(SwitchTargets::SwitchInt(
-                    IntegerTy::Signed(int_ty),
+                    LiteralTy::Char,
                     targets,
                     otherwise,
                 ))
             }
-            LiteralTy::UInt(int_ty) => {
-                let targets: Vec<(ScalarValue, BlockId)> = targets
+            LiteralTy::Int(int_ty) => {
+                let targets: Vec<(Literal, BlockId)> = targets
                     .iter()
                     .map(|(v, tgt)| {
-                        let v = ScalarValue::from_le_bytes(
-                            IntegerTy::Unsigned(int_ty),
-                            v.data_le_bytes,
+                        let v = charon_lib::ast::values::Literal::Scalar(
+                            ScalarValue::from_le_bytes(IntegerTy::Signed(int_ty), v.data_le_bytes),
                         );
                         let tgt = self.translate_basic_block_id(*tgt);
                         (v, tgt)
                     })
                     .collect();
                 let otherwise = self.translate_basic_block_id(*otherwise);
                 Ok(SwitchTargets::SwitchInt(
-                    IntegerTy::Unsigned(int_ty),
+                    LiteralTy::Int(int_ty),
```

Comment:
Don't reconstruct a thing if you already have it around:
```suggestion
                    switch_ty
```